File: android-lib/src/main/kotlin/org/vander/spotifyclient/model/api/NowPlaying.kt
package org.vander.spotifyclient.model.api

data class NowPlaying(
    val trackId: String? = null,
    val uri: String? = null,
    val title: String? = null,
    val artists: List<String> = emptyList(),
    val imageUrl: String? = null,
    val durationMs: Long? = null,
    val progressMs: Long? = null,
    val isPlaying: Boolean = false,
    val shuffle: Boolean? = null,
    val repeat: String? = null,
    val deviceName: String? = null,
    val deviceType: String? = null,
)

File: packages/core-domain-ts/src/lib/playing/NowPlaying.ts
export type RepeatMode = 'off' | 'one' | 'all';

export type NowPlayingCapabilities = Partial<{
  pause: boolean;
  resume: boolean;
  seek: boolean;
  next: boolean;
  previous: boolean;
}>;

export interface NowPlaying {
  // identity
  trackId?: string;
  uri?: string;

  // display
  title?: string;
  artists?: string[];
  imageUrl?: string | null;

  // timing
  durationMs?: number | null;
  progressMs?: number | null;
  isPlaying: boolean;

  // modes
  shuffle?: boolean;
  repeat?: RepeatMode;

  // capabilities for UI buttons
  can?: NowPlayingCapabilities;

  // optional device info for UX
  deviceName?: string;
  deviceType?: string;
}
